src/Silex/CallbackResolver.php

```diff
@@ -33,40 +33,59 @@ public function __construct(Container $app)
      */
     public function isValid($name)
     {
-        return is_string($name) && preg_match(static::SERVICE_PATTERN, $name);
+        return is_string($name) && (preg_match(static::SERVICE_PATTERN, $name) || isset($this->app[$name]));
     }
 
     /**
      * Returns a callable given its string representation.
      *
      * @param string $name
      *
-     * @return array A callable array
+     * @return callable A callable value
      *
      * @throws \InvalidArgumentException In case the method does not exist.
      */
     public function convertCallback($name)
     {
-        list($service, $method) = explode(':', $name, 2);
+        if (preg_match(static::SERVICE_PATTERN, $name)) {
+            list($service, $method) = explode(':', $name, 2);
 
-        if (!isset($this->app[$service])) {
-            throw new \InvalidArgumentException(sprintf('Service "%s" does not exist.', $service));
-        }
+            if (!isset($this->app[$service])) {
+                $this->throwServiceNotExists($service);
```

Comment:
Good point about Pimple already throwing an exception. I was just copying the design of the code that was there before. I'm not a huge fan of rethrowing, but wouldn't mind removing the checks to see if the service actually exists and let pimple throw an error message.